File: Java/BOJ/2669 직사각형 네개의 합집합의 면적 구하기.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[][] graph = new int[101][101];
        for(int t = 0; t < 4; t++) {
        	StringTokenizer st = new StringTokenizer(br.readLine()," ");
        	int x = Integer.parseInt(st.nextToken());
        	int y = Integer.parseInt(st.nextToken());
        	int nx = Integer.parseInt(st.nextToken());
        	int ny = Integer.parseInt(st.nextToken());
        	for(int i = x; i < nx; i++) {
        		for(int j = y; j < ny; j++) {
        			graph[i][j] = 1;
        		}
        	}
        }
        int result = 0;
        for (int i = 0 ; i < 101; i++) {
        	for (int j = 0; j < 101; j++) {
        		if(graph[i][j] == 1) {
        			result++;
        		}
        	}
        }
        System.out.println(result);
    }
}

File: Python/BOJ/2669 직사각형 네개의 합집합의 면적 구하기.py
import sys
graph =[[0]*101 for _ in range(101)]

for _ in range(4):
    x, y, nx, ny = map(int, sys.stdin.readline().split())
    for i in range(x,nx):
        for j in range(y,ny):
            graph[i][j] = 1

result = 0
for i in range(101):
    for j in range(101):
        if graph[i][j] == 1:
            result += 1
print(result)